### PR TITLE
KOGITO-2001: BDD tests: Refactor Deploy Quarkus and Deploy Spring Boo…

### PR DESCRIPTION
**Pull request by Sgitario**
…t and other scenarios to use the domain steps

JIRA Ticket: https://issues.redhat.com/browse/KOGITO-2001
Description: Change the Quarkus and Spring Boot and other scenarios to use the domain steps

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by Sgitario** [comment]
I ran all the features that I modified but the performance feature. All these worked fine for me locally.

File: test/steps/process.go [hunk 1]
```diff
@@ -24,11 +24,12 @@ import (
 
 func registerProcessSteps(s *godog.Suite, data *Data) {
 	s.Step(`^Start "([^"]*)" process on service "([^"]*)" with body:$`, data.startProcessOnService)
+	s.Step(`^Start "([^"]*)" process on service "([^"]*)" within (\d+) minutes with body:$`, data.startProcessOnServiceWithinMinutes)
 	s.Step(`^Service "([^"]*)" contains (\d+) (?:instance|instances) of process with name "([^"]*)"$`, data.serviceContainsInstancesOfProcess)
 	s.Step(`^Service "([^"]*)" contains (\d+) (?:instance|instances) of process with name "([^"]*)" within (\d+) minutes$`, data.serviceContainsInstancesOfProcessWithinMinutes)
 }
 
-func (data *Data) startProcessOnService(processName, serviceName string, body *messages.PickleStepArgument_PickleDocString) error {
+func (data *Data) startProcessOnService(processName, serviceName string, timeoutInMin int, body *messages.PickleStepArgument_PickleDocString) error {
```
Comment (sutaakar) [inline test/steps/process.go:32]: Seems to be a leftover.

Comment (Sgitario) [reply test/steps/process.go:32]: yes, sorry

File: test/features/binary_build.feature [hunk 2]
```diff
@@ -16,7 +16,7 @@ Feature: Use binary build to deploy the service
     And Start build with name "process-springboot-example-binary" from local example service path "process-springboot-example/target"
 
     Then Kogito application "process-springboot-example" has 1 pods running within 5 minutes
-    And HTTP GET request on service "process-springboot-example" with path "orders" is successful within 2 minutes
+    And Service "process-springboot-example" contains 0 instance of process with name "orders" within 2 minutes
```
Comment (radtriste) [inline test/features/binary_build.feature:19]: Could we create a new step for this,  like `Service "process-springboot-example" with process name "orders" is available within 2 minutes` ?
that would the confusion with `0 instance of process`. we just want to know if service is avaible here.

File: test/features/deploy_quarkus_service.feature [hunk 3]
```diff
@@ -60,8 +60,8 @@ Feature: Deploy quarkus service
       | config | native      | <native> |
       | config | persistence | enabled  |
     And Kogito application "process-quarkus-example" has 1 pods running within <minutes> minutes
-    And HTTP GET request on service "process-quarkus-example" with path "orders" is successful within 3 minutes
-    And HTTP POST request on service "process-quarkus-example" with path "orders" and body:
+    And Service "process-quarkus-example" contains 0 instances of process with name "orders" within 3 minutes
+    And Start "orders" process on service "process-quarkus-example" with body:
```
Comment (radtriste) [inline test/features/deploy_quarkus_service.feature:64]: That could be replaced by one step:
`And Start "orders" process on service "process-quarkus-example" within 3 minutes with body:`

File: test/features/deploy_quarkus_service.feature [hunk 4]
```diff
@@ -131,9 +131,9 @@ Feature: Deploy quarkus service
       | config | persistence | enabled  |
       | config | events      | enabled  |
     And Kogito application "process-quarkus-example" has 1 pods running within <minutes> minutes
-    And HTTP GET request on service "process-quarkus-example" with path "orders" is successful within 3 minutes
+    And Service "process-quarkus-example" contains 0 instances of process with name "orders" within 3 minutes
 
-    When HTTP POST request on service "process-quarkus-example" with path "orders" and body:
+    When Start "orders" process on service "process-quarkus-example" with body:
```
Comment (radtriste) [inline test/features/deploy_quarkus_service.feature:136]: could be replaced by:
`When Start "orders" process on service "process-quarkus-example" within 3 minutes with body:`

File: test/features/deploy_springboot_service.feature [hunk 5]
```diff
@@ -21,8 +21,8 @@ Feature: Deploy spring boot service
     And Deploy springboot example service "process-springboot-example" with configuration:
       | config | persistence | enabled |
     And Kogito application "process-springboot-example" has 1 pods running within 10 minutes
-    And HTTP GET request on service "process-springboot-example" with path "orders" is successful within 3 minutes
-    And HTTP POST request on service "process-springboot-example" with path "orders" and body:
+    And Service "process-springboot-example" contains 0 instances of process with name "orders" within 3 minutes
+    And Start "orders" process on service "process-springboot-example" with body:
```
Comment (radtriste) [inline test/features/deploy_springboot_service.feature:25]: Same here can be merged

File: test/features/deploy_quarkus_service.feature (hunk 4 above)
Comment (Sgitario) [reply test/features/deploy_quarkus_service.feature]: Done

File: test/features/binary_build.feature (hunk 2 above)
Comment (Sgitario) [reply test/features/binary_build.feature]: Done

File: test/features/deploy_quarkus_service.feature (hunk 3 above)
Comment (Sgitario) [reply test/features/deploy_quarkus_service.feature]: Done

File: test/features/deploy_springboot_service.feature (hunk 5 above)
Comment (Sgitario) [reply test/features/deploy_springboot_service.feature]: Done

File: test/features/deploy_springboot_service.feature [hunk 6]
```diff
@@ -71,9 +71,9 @@ Feature: Deploy spring boot service
       | config | persistence | enabled  |
       | config | events      | enabled  |
     And Kogito application "process-springboot-example" has 1 pods running within 10 minutes
-    And HTTP GET request on service "process-springboot-example" with path "orders" is successful within 3 minutes
+    And Service "process-springboot-example" contains 0 instance of process with name "orders" within 3 minutes
 
-    When HTTP POST request on service "process-springboot-example" with path "orders" and body:
+    When Start "orders" process on service "process-springboot-example" with body:
```
[the remark this answers is not in the document]
Comment (Sgitario) [reply test/features/deploy_springboot_service.feature]: Done

File: test/steps/process.go [hunk 7]
```diff
@@ -41,6 +44,49 @@ func (data *Data) startProcessOnService(processName, serviceName string, body *m
...
+		return err
+	}
+
+	return nil
+}
+
+func (data *Data) serviceWithProcessNameIsAvailableWithinMinutes(serviceName, processName string, timeoutInMin int) error {
+	return framework.WaitForOnOpenshift(data.Namespace, fmt.Sprintf("Process %s is not available yet", processName), timeoutInMin,
+		func() (bool, error) {
+			err := data.serviceWithProcessNameIsAvailable(serviceName, processName)
```
Comment (radtriste) [inline test/steps/process.go:81]: Here we are going to get each time the route URI (and wait for it). is that expected ?
I think we should get it once before the wait and then just call the `framework.GetProcessInstances` method

Comment (Sgitario) [reply test/steps/process.go]: I wanted to reuse the first method and thought getting each time the route it was not a big issue...
Changed.

**Pull request merged**